### PR TITLE
clean up agent values - fix diagnostic pod logic

### PR DESCRIPTION
**Pull request by JesseGoodier**
## What does this PR change?
- update hosted agent to run diagnostics with thanos
- clean up agent values 

## Does this PR rely on any other PRs?
No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)
Fixed logic in diagnostic pod deployment

## What risks are associated with merging this PR? What is required to fully test this PR?
Unknown configs, though I believe this risk is low.

## How was this PR tested?
Local install, will be sharing with a customer asap.

## Have you made an update to documentation? If so, please provide the corresponding PR.
NA

### CODE REVIEW

**Review by JesseGoodier — COMMENTED** [review]
/cherry-pick v1.108

**Review by chipzoller — APPROVED** [review]

File: cost-analyzer/templates/diagnostics-deployment.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.diagnostics.enabled }}
-{{- if or .Values.global.thanos.enabled (not (empty .Values.kubecostModel.federatedStorageConfigSecret )) -}}
+{{- if or ( not (empty .Values.prometheus.server.extraSecretMounts )) (not (empty .Values.kubecostModel.federatedStorageConfigSecret )) -}}
```
Comment (thomasvn) [inline cost-analyzer/templates/diagnostics-deployment.yaml:2]: I don't fully understand this change. Are we assuming that when users are running Thanos, they will always have the `.Values.prometheus.server.extraSecretMounts` set? If so we should leave a comment here.

Comment (jessegoodier) [reply cost-analyzer/templates/diagnostics-deployment.yaml:2]: That is correct.

No doubt there's a better way to do this. Probably check if the list of sidecarContainers has one with an image containing thanos?

Though I'd also consider forcing thanos configs to add a flag to enable diagnostics.

I was just following the logic of the federated store. 

Here are the values for a thanos agent:

```yaml
prometheus:
  server:
    enableAdminApi: true
    extraArgs:
      storage.tsdb.max-block-duration: 2h
      storage.tsdb.min-block-duration: 2h
      storage.tsdb.retention: 2w
    extraVolumes:
    - name: object-store-volume
      secret:
        secretName: kubecost-thanos
    global:
      external_labels:
        cluster_id: kc-demo-dev
    sidecarContainers:
    - args:
      - sidecar
      - --log.level=debug
      - --tsdb.path=/data/
      - --prometheus.url=http://127.0.0.1:9090
      - --objstore.config-file=/etc/config/object-store.yaml
      - --min-time=-3h
      image: thanosio/thanos:v0.29.0
      name: thanos-sidecar
```

**Comment by chipzoller** [comment]
This doesn't look like it got cherry picked.

File: cost-analyzer/templates/diagnostics-deployment.yaml (hunk 1 above)
Comment (thomasvn) [reply cost-analyzer/templates/diagnostics-deployment.yaml:2]: Gotcha! Was there any issue with `.Values.global.thanos.enabled` though?

Comment (jessegoodier) [reply cost-analyzer/templates/diagnostics-deployment.yaml:2]: yes, it isn't required for agents, so they wouldn't ship diagnostics by default.
https://github.com/kubecost/poc-common-configurations/blob/main/aws/values-amazon-secondary.yaml

**Pull request merged**